pkg/controller/controller.go

```diff
@@ -1015,3 +1019,50 @@ func (c *Controller) GetReceiverTmpl(cm *v2beta2.ConfigmapKeySelector) (*templat
 
 	return globalTmpl.ParserText(text...)
 }
+
+type clusterConfig struct {
+	MultiCluster multiCluster `json:"multicluster,omitempty" yaml:"multicluster,omitempty"`
+}
+
+type multiCluster struct {
+	ClusterName string `json:"clusterName,omitempty" yaml:"clusterName,omitempty"`
+}
+
+// GetCluster returns the name of the cluster where the notification manager is located.
+// If a cluster name is specified, the specified cluster name is used.
+// Otherwise, for kubespere platforms (version >= 3.3), it will try to get the cluster name
+// from the configmap kubesphere-config.
+// Otherwise, the default cluster name "default" will be returned.
+func (c *Controller) GetCluster() string {
```

Comment:
Can we make this GetCluster irrelevant to kubesphere?
For example, if a non-kubesphere cluser set a clustername in a configmap in the same format as kubesphere's cm, it can use the same cluster-setting function.

Reply:
The way to get the cluster name from the configmap should not be a general way, we can discuss how to make it more general after we have the requirement

Reply:
In nm code, logic related to kubesphere is limited to the sidecar.
This PR is actually hardcoded kubesphere in nm code.

It does not affect the functionality, just kind of design stuff
We can leave it as it is for now.